### PR TITLE
[Solved #49] Implement socket auth

### PR DESCRIPTION
**Pull request by supunTE**
Socket auth implemented on the server.
Update client-side socket communication to receive and send auth token(if any)

### CODE REVIEW

File: apps/client/src/services/initialize-socket.ts [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
+let authToken = '';
+
+const socket = io(SOCKET_ENDPOINT, {
+  auth: {
+    token: authToken,
+  },
+  autoConnect: false,
   withCredentials: true,
-});
+}) as Socket & { auth: { token: AuthToken } };
```
Comment (dalanad) [inline apps/client/src/services/initialize-socket.ts:18]: Shall we separate this type delerations

**Review by dalanad — CHANGES_REQUESTED** [review]

File: apps/client/src/services/initialize-socket.ts (hunk 1 above)
Comment (supunTE) [reply apps/client/src/services/initialize-socket.ts:18]: This is fixed in commit 4564f6717a440f69a2531402a290b8ef92b21055 under PR #73 
https://github.com/the-ai-team/razor/pull/73/files#diff-e7af1152c8e7f0a3b83f78d4ab66c7a8368764cc1b31f300ac8c5e8569602c1b

FYI: `initialize-socket` was renamed as `socket-communication` in same PR.

**Review by dalanad — APPROVED** [review]

File: apps/client/src/pages/home/Home.page.tsx [hunk 2]
```diff
@@ -17,23 +16,34 @@ import {
 } from '../../components';
 
 export function Home(): ReactElement {
-  const { id } = useParams();
+  const { roomId } = useParams();
+  // disconnect any socket connection if user navigates to home page.
+  endSocket();
```
Comment (supunTE) [inline apps/client/src/pages/home/Home.page.tsx:21]: This was changed from an upcoming PR #73  66ea107312d0f51fe1fb478601b601bc0b570d45

**Comment by supunTE** [comment]
Related #49

**Pull request merged**